library/Barberry/Cache.php

```diff
@@ -29,45 +27,45 @@ public function save($streamOrContent, Request $request): string
 
     public function invalidate($id): void
     {
-        $dir = $this->path . nonlinear\generateDestination($id);
-        if (is_dir($dir)) {
-            fs\rmDirRecursive($dir);
+        $path  = nonlinear\generateDestination($id);
+        if ($this->directoryExists($path)) {
+            $this->filesystem->deleteDirectory($path);
         }
     }
 
     private function writeToFilesystem($streamOrContent, $filePath): void
     {
-        if (!is_dir($d = dirname($filePath))) {
-            if (!mkdir($d, 0777, true) && !is_dir($d)) {
-                throw new \RuntimeException(sprintf('Directory "%s" was not created', $d));
-            }
+        $dir = dirname($filePath);
+        if (!$this->directoryExists($dir)) {
+            $this->filesystem->createDirectory($dir);
```

Comment:
@antonsacred please make sure that the mode for new directory is `0777` to make it follow the umask of the process.